### PR TITLE
Clarify mesos vs marathon health checks

### PR DESCRIPTION
**Pull request by sascala**
https://jira.mesosphere.com/browse/MARATHON-7557

### CODE REVIEW

File: docs/docs/health-checks.md [hunk 1]
```diff
@@ -16,9 +16,123 @@ application's tasks.
...
+- `COMMAND`
+- `MESOS_HTTP`
+- `MESOS_HTTPS`
+- `MESOS_TCP`
+
+### Marathon-level health checks
+
+Marathon-level health checks (HTTP, HTTPS, and TCP) are executed by Marathon and thus test reachability from the current Marathon leader. Marathon-level health checks have several limitations:
+
+- The API for health checks varies from one framework to the next and common functionality must be reimplemented with each new framework that requires health checks.
```
Comment (timcharper) [inline docs/docs/health-checks.md]: This sounds like potentially a comment for framework developers and not users of Marathon. Does that sound right to you?

File: docs/docs/health-checks.md [hunk 2]
```diff
@@ -16,9 +16,123 @@ application's tasks.
...
+These limitations may be acceptable for smaller clusters with low-scale tasks, but Marathon-level health checks should not be used on large clusters with highly scaled Marathon tasks.
+
+**Important:** Marathon-based health checks are deprecated and will be removed in a future version. See the [1.4.0 release notes](https://github.com/mesosphere/marathon/releases/tag/v1.4.0) for more information.
+
+### Mesos-level health checks
+
+Mesos-level health checks (`MESOS_HTTP`, `MESOS_HTTPS`, `MESOS_TCP`, and `COMMAND`) health checks are locally
+executed by Mesos on the agent running the corresponding task and thus test reachability from the Mesos executor. Mesos-level health checks offer the following advantages over Marathon-level health checks:
+
+- All built-in executors support native health checks via the new API and library. Custom executors can now use the same library and API to implement health checks, regardless of whether their tasks are actual processes or something else.
```
Comment (timcharper) [inline docs/docs/health-checks.md]: I also wonder if this bit is for a different audience

File: docs/docs/health-checks.md [hunk 3]
```diff
@@ -16,9 +16,123 @@ application's tasks.
...
+
+- Mesos-level health checks are performed as close to the task as possible, so they are are not affected by networking failures.
+
+- Mesos-level health checks delegated to the agents running the tasks, so the number of tasks that can be health checked can scale horizontally with the number of agents in the cluster.
```
Comment (timcharper) [inline docs/docs/health-checks.md]: health check *are* delegated

Comment (timcharper) [inline docs/docs/health-checks.md]: Should there be a `-` when using health check as a verb?

> so the number of tasks that can be health-checked

I'm not a pro tech writer, maybe it is fine as is :) this just seemed strange to me.

File: docs/docs/health-checks.md [hunk 4]
```diff
@@ -16,9 +16,123 @@ application's tasks.
...
+
+#### Limitations and considerations
+
+- Mesos-level health checks consume extra resources on the agents; moreover, there is some overhead for fork-execing a process and entering the tasks’ namespaces every time a task is checked.
```
Comment (timcharper) [inline docs/docs/health-checks.md]: I understand there are plans to improve the efficiency here. Confirm with Mesos team? If so, perhaps:

> there is (currently) some overhead

File: docs/docs/health-checks.md [hunk 5]
```diff
@@ -16,9 +16,123 @@ application's tasks.
...
+
+- The health check processes share resources with the task that they check. Your application definition must account for the extra resources consumed by the health checks.
+
+- Mesos-level health checks require tasks to listen on the loopback interface in addition to whatever interface they require. If you run a service in production, you will want to make sure that the users can reach it.
```
Comment (timcharper) [inline docs/docs/health-checks.md]: The loopback interface in a container is not the same as the host, unless if the container is not using network isolation.

As a user, I might be confused as to whether it needs to listen on the hosts loopback interface. Do you think this is a worth-while clarification?

> on the *container's* loopback interface

File: docs/docs/health-checks.md [hunk 6]
```diff
@@ -16,9 +16,123 @@ application's tasks.
...
+
+#### `COMMAND` health checks
+
+You must escape any double quotes in your commands. This is required because Mesos runs the healthcheck command inside via `/bin/sh -c ""`.
+  See the example below and [MESOS-4812](https://issues.apache.org/jira/browse/MESOS-4812) for details.
```
Comment (timcharper) [inline docs/docs/health-checks.md]: Ugh. This is awful. Good to mention.

File: docs/docs/health-checks.md (hunk 1 above)
Comment (sascala) [inline docs/docs/health-checks.md]: Yeah... this might be more relevant to Mesos documentation

File: docs/docs/health-checks.md (hunk 2 above)
Comment (sascala) [inline docs/docs/health-checks.md]: I see your point

File: docs/docs/health-checks.md (hunk 3 above)
Comment (sascala) [inline docs/docs/health-checks.md]: I probably shouldn't be verbifying the word in the first place.

File: docs/docs/health-checks.md (hunk 4 above)
Comment (sascala) [inline docs/docs/health-checks.md]: 👍

File: docs/docs/health-checks.md (hunk 5 above)
Comment (sascala) [inline docs/docs/health-checks.md]: yes :-)

**Pull request merged**